Libraries/LibGUI/Menu.h

```diff
@@ -56,19 +56,20 @@ class Menu final : public Core::Object {
     void add_separator();
     Menu& add_submenu(const String& name);
 
-    void popup(const Gfx::IntPoint& screen_position);
+    void popup(const Gfx::IntPoint& screen_position, Action* default_action = nullptr);
     void dismiss();
 
 private:
     friend class MenuBar;
 
-    int realize_menu();
+    int realize_menu(Action* default_action = nullptr);
     void unrealize_menu();
-    void realize_if_needed();
+    void realize_if_needed(Action* default_action);
 
     int m_menu_id { -1 };
     String m_name;
     NonnullOwnPtrVector<MenuItem> m_items;
+    Action* m_last_default_action { nullptr };
```

Comment:
Using a raw pointer like this is pretty sloppy. Let's use a `WeakPtr<Action>` instead :)

Reply:
Well, the pointer is only ever compared to determine a match, the object is never accessed using it. Still want me to change it to a WeakPtr instead?

Reply:
Ok, changed it to a WeakPtr